From 369b82556e9b2fa5f87e82638053c2ec1c54ba56 Mon Sep 17 00:00:00 2001
From: Nataraj Deshpande <nataraj.deshpande@intel.com>
Date: Tue, 14 Apr 2020 14:56:20 -0700
Subject: [PATCH] Limit vulkan version to 1.1 for Android

Current Android dessert versions such as Pie, Q reject
vulkan version > 1.1. Restrict the versions to 1.1 for
platforms running these Android desserts.

Change-Id: I7145cc6283f32263261f4420a2ea0a8b0db4ddc9
---
 src/intel/vulkan/anv_extensions_gen.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/intel/vulkan/anv_extensions_gen.py b/src/intel/vulkan/anv_extensions_gen.py
index e83c0e16958..aacdfdde776 100644
--- a/src/intel/vulkan/anv_extensions_gen.py
+++ b/src/intel/vulkan/anv_extensions_gen.py
@@ -167,6 +167,12 @@ anv_physical_device_api_version(struct anv_physical_device *device)
         return version;
     version = ${version.version.c_vk_version()};
 
+#if defined(ANDROID_API_LEVEL) && ANDROID_API_LEVEL <= 30
+    /* Restrict the vulkan version for current versions such as
+     * Android Pie, Q.
+     */
+    version = MIN2(version, VK_MAKE_VERSION(1, 1, 0));
+#endif
 %endfor
     return version;
 }
-- 
2.24.1

